tests/test-cases/Group6-VIC-Machine/6-17-Configure-TLS.robot

```diff
@@ -0,0 +1,168 @@
+
+*** Keywords ***
+Setup Test Environment
+    Set Test Environment Variables
+    Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword And Ignore Error  Cleanup Datastore On Test Server
+
+    ${output}=  Run  bin/vic-machine-linux create ${vicmachinetls} --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --image-store=%{TEST_DATASTORE} --bridge-network=%{BRIDGE_NETWORK} --public-network=%{PUBLIC_NETWORK} --cert-path=${EXECDIR}/foo-bar-certs/
+    Should Contain  ${output}  --tlscacert="${EXECDIR}/foo-bar-certs/ca.pem" --tlscert="${EXECDIR}/foo-bar-certs/cert.pem" --tlskey="${EXECDIR}/foo-bar-certs/key.pem"
+    Should Contain  ${output}  Generating CA certificate/key pair - private key in ${EXECDIR}/foo-bar-certs/ca-key.pem
+    Should Contain  ${output}  Generating server certificate/key pair - private key in ${EXECDIR}/foo-bar-certs/server-key.pem
+    Should Contain  ${output}  Generating client certificate/key pair - private key in ${EXECDIR}/foo-bar-certs/key.pem
+    Should Contain  ${output}  Generated browser friendly PFX client certificate - certificate in ${EXECDIR}/foo-bar-certs/cert.pfx
+
+    Should Contain  ${output}  Installer completed successfully
+    Get Docker Params  ${output}  ${true}
+
+    ${save_env}=  Run  cat ${EXECDIR}/foo-bar-certs/%{VCH-NAME}.env
+    Should Contain  ${save_env}  DOCKER_CERT_PATH=${EXECDIR}/foo-bar-certs
+    Log To Console  Installer completed successfully: %{VCH-NAME}
+
+Cleanup
+    Run  rm -rf bundle; rm -rf foo-bar-certs
+    Run Keyword  Cleanup VIC Appliance On Test Server
+
+*** Test Cases ***
+Configure VCH - Server cert with untrusted CA
+    ${domain}=  Get Environment Variable  DOMAIN  ''
+    Run Keyword If  '${domain}' == ''  Pass Execution  Skipping test - domain not set, won't generate keys
+    # Generate CA and wildcard cert for *.<DOMAIN>
+    Generate Certificate Authority
+    Generate Wildcard Server Certificate
+
+    ${out}=  Run  cp /root/ca/cert-bundle.tgz .; tar xvf cert-bundle.tgz
+    Log  ${out}
+
+    # Run vic-machine configure, supply server cert and key
+    ${output}=  Run  bin/vic-machine-linux configure --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --key "bundle/*.${domain}.key.pem" --cert "bundle/*.${domain}.cert.pem" ${vicmachinetls} --cert-path "out-bundle" --debug 1
+    Log  ${output}
+    Should Contain  ${output}  Completed successfully
+
+    # Verify that the supplied certificate is presented on web interface
+    ${output}=  Run  openssl s_client -showcerts -connect %{VCH-IP}:2378
+    Log  ${output}
+    Should Contain  ${output}  issuer=/C=US/ST=California/L=Los Angeles/O=Stark Enterprises/OU=Stark Enterprises Certificate Authority/CN=Stark Enterprises Global CA
+
+    Run  rm -rf bundle
+    Run  rm -f cert-bundle.tgz
+    Run  rm -f out-bundle
+    Run  rm -rf /root/ca
```

Comment:
I think what you actually want here is Suite Teardown   to be VCH Cleanup and take all these rm's and move them to the cleanup keyword and make it a Test Teardown.